WordPress/src/main/java/org/wordpress/android/ui/prefs/notifications/NotificationsSettingsActivity.kt

```diff
@@ -0,0 +1,149 @@
+    private lateinit var fragmentContainer: View
+
+    public override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(layout.notifications_settings_activity)
+        fragmentContainer = findViewById(id.fragment_container)
+
+        // Get shared preferences for main switch.
+        sharedPreferences = PreferenceManager.getDefaultSharedPreferences(this@NotificationsSettingsActivity)
+
+        // Set up primary toolbar
+        setUpToolbar()
+
+        // Set up main switch
+        setUpMainSwitch()
+
+        if (savedInstanceState == null) {
+            @Suppress("DEPRECATION")
+            fragmentManager.beginTransaction()
+                    .add(id.fragment_container, NotificationsSettingsFragment())
+                    .commit()
+        }
+
+        messageContainer = findViewById(id.notifications_settings_message_container)
+        messageTextView = findViewById(id.notifications_settings_message)
+    }
+
+    override fun onStop() {
+        EventBus.getDefault().unregister(this)
+        super.onStop()
+    }
+
+    override fun onStart() {
+        super.onStart()
+        EventBus.getDefault().register(this)
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        when (item.itemId) {
+            android.R.id.home -> {
+                onBackPressed()
+                return true
+            }
+        }
+        return super.onOptionsItemSelected(item)
+    }
+
+    @Subscribe(threadMode = MAIN)
+    fun onEventMainThread(event: NotificationsSettingsStatusChanged) {
+        if (TextUtils.isEmpty(event.message)) {
+            messageContainer.visibility = View.GONE
+        } else {
+            messageContainer.visibility = View.VISIBLE
+            messageTextView.text = event.message
+        }
+    }
+
+    /**
+     * Set up primary toolbar for navigation and search
+     */
+    private fun setUpToolbar() {
+        val toolbar = findViewById<Toolbar>(id.toolbar_with_search)
+
+        toolbar?.let { setSupportActionBar(it) }
+
+        supportActionBar?.apply {
+            setTitle(string.notification_settings)
+            setDisplayHomeAsUpEnabled(true)
+            setDisplayShowTitleEnabled(true)
+        }
+    }
+
+    /**
+     * Sets up main switch to disable/enable all notification settings
+     */
+    private fun setUpMainSwitch() {
+        val mainSwitchToolBarView = findViewById<PrefMainSwitchToolbarView>(id.main_switch)
+        mainSwitchToolBarView.setMainSwitchToolbarListener(this)
+
+        // Set main switch state from shared preferences.
+        val isMainChecked = sharedPreferences.getBoolean(getString(string.wp_pref_notifications_main), true)
+        mainSwitchToolBarView.loadInitialState(isMainChecked)
+        hideDisabledView(isMainChecked)
+    }
+
+    override fun onMainSwitchCheckedChanged(buttonView: CompoundButton?, isChecked: Boolean) {
```

Comment:
This was removed in a previous PR right?

Reply:
No, `onMainSwitchCheckedChanged` was being called unnecessarily at launch. I fixed that in #17498. It didn't change this class.